pkg/db/queries.sql

```diff
@@ -75,3 +75,56 @@ FROM
 GROUP BY
 	originator_node_id;
 
+-- name: GetAddressLogs :many
+SELECT
+	a.address,
+	encode(a.inbox_id, 'hex') AS inbox_id,
+	a.association_sequence_id
+FROM
+	address_log a
+	INNER JOIN (
+		SELECT
+			address,
+			MAX(association_sequence_id) AS max_association_sequence_id
+		FROM
+			address_log
+		WHERE
+			address = ANY (@addresses::TEXT[])
+			AND revocation_sequence_id IS NULL
+		GROUP BY
+			address) b ON a.address = b.address
+	AND a.association_sequence_id = b.max_association_sequence_id;
+
+-- name: InsertAddressLog :one
+INSERT INTO address_log(address, inbox_id, association_sequence_id, revocation_sequence_id)
+	VALUES (@address, decode(@inbox_id, 'hex'), @association_sequence_id, @revocation_sequence_id)
+RETURNING
+	*;
```

Comment:
Could we follow `idempotent inserts` from https://github.com/xmtp/xmtpd/pull/113, so that all inserts have consistent behavior? I think that also gives us a way to address the scenario where this row already exists

Reply:
Done. We don't even look at the returned value, so I also dropped the `RETURNING` clause